### PR TITLE
Plans: Split jetpack-plans 'utils' into separate files

### PR DESCRIPTION
**Pull request by mattgawarecki**
**NOTE:** This change is _optional_ and subject to team feedback! It's not fully polished and could probably use some better file organization, but I'm interested to know what people think.

#### Purpose of this Pull Request

"Utils" files are common in lots of software projects, but over time they tend to accumulate lots of miscellaneous functions, hide complexity, and make it harder to understand the flow of dependencies across the codebase. This PR splits up one such example at `client/my-sites/plans/jetpack-plans/utils.ts` and moves its functions into either (a) their own separate files, or (b) directly into the files that call them.

Separating files like this should also theoretically help with page load performance, because the build process will be able to make more concise bundles. Whereas importing even one function from `utils.ts` file would also bring with it several React components, importing one function from a file like `get-highest-annual-discount.ts` requires fewer dependencies and hence less JS for the browser to load.

#### Changes proposed

* Split out all the methods in `my-sites/plans/jetpack-plans/util.ts`.
* Delete `my-sites/plans/jetpack-plans/util.ts`, since it's no longer necessary.

#### Testing instructions

* Verify all unit tests still pass (`yarn test`).
* Verify all E2E tests still pass.
* Visit the Plans/Pricing page in Calypso Blue, Calypso Green, and Jetpack Connect.
* Verify that the page still looks and behaves exactly as it does in production for all scenarios.

### CODE REVIEW

File: client/my-sites/plans/jetpack-plans/plans-filter-bar/get-highest-annual-discount.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+	// Protect against null values and division by zero
+	if ( ! yearly || ! monthly ) {
+		return null;
+	}
+
+	const monthlyCostPerYear = monthly * 12;
+	const yearlySavings = monthlyCostPerYear - yearly;
+
+	return yearlySavings / monthlyCostPerYear;
+};
```
Comment (mattgawarecki) [inline client/my-sites/plans/jetpack-plans/plans-filter-bar/get-highest-annual-discount.ts:29]: This function is the only change in logic inside this refactor PR -- I moved the logic to calculate the discount for an individual slug into its own function, for readability sake.

**Comment by monsieur-z** [comment]
@mattgawarecki, once test is failing (`getHighestAnnualDiscount › returns null if highest discount would round to 0%`).

**Review by monsieur-z — CHANGES_REQUESTED** [review]
I believe the `client/my-sites/plans/jetpack-plans/checkout.ts` requires the `page` module to be imported.

**Review by monsieur-z — APPROVED** [review]
Code changes look good. The pricing page tested well in all 3 environments, and all unit tests pass. There's a failing e2e test on mobile, but which doesn't seem related to this PR.

Thanks for taking care of that!

**Pull request merged**